pkg/controller/user/user_controller.go

```diff
@@ -531,6 +570,74 @@ func (c *Controller) deleteRoleBindings(user *iamv1alpha2.User) error {
 	return nil
 }
 
+func (c *Controller) deleteLoginRecords(user *iamv1alpha2.User) error {
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}).String(),
+	}
+	deleteOptions := metav1.NewDeleteOptions(0)
+
+	if err := c.ksClient.IamV1alpha2().LoginRecords().
+		DeleteCollection(deleteOptions, listOptions); err != nil {
+		klog.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) syncUserStatus(user *iamv1alpha2.User) (*iamv1alpha2.User, error) {
+	records, err := c.loginRecordLister.List(labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}))
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+	now := time.Now()
+	attempts := 0
+	for _, record := range records {
+		if record.Spec.Type == iamv1alpha2.LoginFailure &&
+			now.Sub(record.CreationTimestamp.Time) <= c.authenticationOptions.AuthenticateRateLimiterDuration {
+			attempts++
+		}
+	}
+	if user.Status.State != iamv1alpha2.UserDisabled {
+		expect := user.DeepCopy()
+		// UserAuthLimitExceeded
+		if attempts >= c.authenticationOptions.AuthenticateRateLimiterMaxTries {
+			expect.Status = iamv1alpha2.UserStatus{
+				State:              iamv1alpha2.UserAuthLimitExceeded,
+				Reason:             "auth rate limit exceeded",
+				LastTransitionTime: &metav1.Time{Time: time.Now()},
+			}
+		} else if user.Status.LastTransitionTime != nil &&
+			time.Now().Sub(user.Status.LastTransitionTime.Time) > c.authenticationOptions.AuthenticateRateLimiterDuration {
+			// UserActive
+			if encrypted, _ := strconv.ParseBool(user.Annotations[iamv1alpha2.PasswordEncryptedAnnotation]); encrypted &&
```

Comment:
don't bother to parseBool, just do `user.Annotations[iamv1alpha2.PasswordEncryptedAnnotation] == "true"`